multi/imxrt-multi/uart.c

```diff
@@ -235,7 +231,7 @@ static void set_baudrate(void *_uart, speed_t baud)
 	uint32_t reg, t;
 	uart_t *uartptr = (uart_t *)_uart;
 
-	reg = calculate_baudrate(baud);
+	reg = calculate_baudrate((uint32_t)libtty_baudrate_to_int(baud));
```

Comment:
What if `libtty_baudrate_to_int(baud)` return `-1`, or `0` ?

Reply:
An update: 
I see that `libtty` does not check the case for `B0` and any `unknown` baud value that is passed to `set_baudrate` callback:
https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/05e9b3785492e1e361a9e8b9e6714b4ff18e2a24/tty/libtty/libtty.c#L443-L448
@agkaminski maybe either  the `set_baudrate` here in `imxrt-multi` for the case `baud <= 0` shall be kind of `no-op` (return;), because of undefined behaviour or the `libtty` should return error for the `ioctl` (see [`man termios`](https://man.openbsd.org/tcsetattr.3#GETTING_AND_SETTING_THE_BAUD_RATE) ).

Reply:
Hmm: The zero baud rate, B0, is used to terminate the connection.  If
       B0 is specified, the modem control lines shall no longer be
       asserted.  Normally, this will disconnect the line.  CBAUDEX is a
       mask for the speeds beyond those defined in POSIX.1 (57600 and
       above).  Thus, B57600 & CBAUDEX is nonzero.

So it seems it should be no-op, but I don't know how to handle this

Reply:
@MaciejPurski, for me, the PR works ok on `imxrt117x`, so it can be mergerd now. Marking as resolved.